index.js

```diff
@@ -180,72 +180,89 @@
   function implPath(path) {
-    return _funcPath(false, path, implementations);
+    var x = implementations, len = path.length;
+    for (var idx = 0; idx < len && has(path[idx], x); idx += 1) {
+      x = x[path[idx]];
+    }
+    return typeof x === 'function' ? x : null;
   }
 
   //  functionName :: Function -> String
-  var functionName = 'name' in function f() {} ?
+  var functionName = 'name' in identity ?
     function functionName(f) { return f.name; } :
     /* istanbul ignore next */
     function functionName(f) {
       var match = /function (\w*)/.exec(f);
       return match == null ? '' : match[1];
     };
 
+  function getBoundFunction(_name) {
+    var name = Prefix + _name;
+    return function(typeRep) {
+      var f = funcPath([name], typeRep);
+      return f == null && typeof typeRep === 'function' ?
+        implPath([functionName(typeRep), name]) :
+        f;
+    };
+  }
+
+  function getBoundMethod(_name) {
+    var name = Prefix + _name;
+    return function(x) {
+      var isPrototype = x != null &&
+                        x.constructor != null &&
+                        x.constructor.prototype === x;
+      var m = null;
+      if (!isPrototype) m = funcPath([name], x);
+      if (m == null)    m = implPath([type(x), 'prototype', name]);
+      return m && m.bind(x);
+    };
+  }
+
   //  $ :: (String, Array TypeClass, StrMap (Array Location)) -> TypeClass
   function $(_name, dependencies, requirements) {
-    function getBoundMethod(_name) {
-      var name = 'fantasy-land/' + _name;
-      return requirements[_name] === Constructor ?
-        function(typeRep) {
-          var f = funcPath([name], typeRep);
-          return f == null && typeof typeRep === 'function' ?
-            implPath([functionName(typeRep), name]) :
-            f;
-        } :
-        function(x) {
-          var isPrototype = x != null &&
-                            x.constructor != null &&
-                            x.constructor.prototype === x;
-          var m = null;
-          if (!isPrototype) m = funcPath([name], x);
-          if (m == null)    m = implPath([type(x), 'prototype', name]);
-          return m && m.bind(x);
-        };
-    }
-
     var name = 'sanctuary-type-classes/' + _name;
     var keys = Object.keys(requirements);
 
+    var functionKeys = keys.filter(function(k) {
+      return requirements[k] === Constructor;
+    });
+
+    var methodKeys = keys.filter(function(k) {
+      return requirements[k] === Value;
+    });
+
     var typeClass = TypeClass(name, dependencies, function(x) {
-      return keys.every(function(_name) {
-        var arg = requirements[_name] === Constructor ? x.constructor : x;
-        return getBoundMethod(_name)(arg) != null;
-      });
+      return functionKeys.every(function(k) {
+               return typeClass.functions[k](x.constructor) != null;
+             }) && methodKeys.every(function(k) {
+               return typeClass.methods[k](x) != null;
+             });
```

Comment:
Let's add `--rule 'indent: [off]'` to the makefile. `indent` is one rule I'm considering disabling in sanctuary-style as it produces so many false positives.